projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts

```diff
@@ -32,21 +32,116 @@ import { VirtualHelperComponent } from './virtual.helper.component';
 
 @Directive({ selector: '[igxFor][igxForOf]' })
 export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestroy {
-    @Input() public igxForOf: any[];
-    @Input() public igxForScrollOrientation: string;
-    @Input() public igxForScrollContainer: any;
-    @Input() public igxForContainerSize: any;
-    @Input() public igxForItemSize: any;
+
+    /**
+     * An @Input property that sets the data to be rendered.
+     * ```html
+     * <ng-template igxFor let-item [igxForOf]="data"></ng-template>
+     * ```
+     */
+    @Input()
+    public igxForOf: any[];
+
+    /**
+     * An @Input property that specifies the scroll orientation.
+     * Only `vertical` and `horizontal` are valid.
```

Comment:
@alexalex972 Suggestion: "Scroll orientation can be `"vertical"` or `"horizontal"`.